src/config/config.rs

```diff
@@ -113,6 +113,31 @@ impl Config {
     pub fn contains(&self, name: &str) -> bool {
         self.types.contains_key(name) || self.unions.contains_key(name)
     }
+
+    pub fn get_all_used_type_names(&self) -> HashSet<String> {
+        let mut set = HashSet::new();
+        let mut stack = vec!["Query".into(), "Mutation".into()];
+        while let Some(type_name) = stack.pop() {
+            if let Some(typ) = self.types.get(&type_name) {
+                set.insert(type_name);
+                for field in typ.fields.values() {
+                    stack.extend(field.args.values().map(|arg| arg.type_of.clone()));
+                    stack.push(field.type_of.clone());
+                }
+            }
+        }
+
+        set
+    }
+
+    pub fn remove_unused_types(&mut self) {
```

Comment:
```suggestion
    pub fn remove_unused_types(self) -> Self {
```

Add documentation to the API.